assets/migrations/postgres/004_new_user_parts_columns.sql

```diff
@@ -0,0 +1,114 @@
+-- +goose NO TRANSACTION
+-- This migration adds 3 more columns to the tuple and changelog tables, and fills past and future data with a trigger.
+-- This migration also adds more indexes to the tuple table. It does so concurrently so that the main table isn't locked
+
+-- +goose Up
+
+ALTER TABLE tuple
+    ADD COLUMN IF NOT EXISTS user_object_type TEXT,
+    ADD COLUMN IF NOT EXISTS user_object_id TEXT,
+    ADD COLUMN IF NOT EXISTS user_relation TEXT;
+
+-- jon becomes (..., jon, ...)
+-- user:jon becomes (user, jon, ...)
+-- team:* becomes (team, *, ...)
+-- group:eng#member becomes (group, eng, member)
+-- +goose StatementBegin
+CREATE OR REPLACE FUNCTION set_new_columns() RETURNS TRIGGER AS $$
+BEGIN
+        NEW.user_object_type = (CASE
+                                    WHEN position(':' in NEW._user) > 0 THEN split_part(NEW._user, ':', 1)
+                                    ELSE ''
+            END),
+
+        NEW.user_object_id = (CASE
+                                  WHEN position('#' in NEW._user) > 0 THEN split_part(split_part(NEW._user, ':', -1), '#', 1)
+                                  ELSE split_part(NEW._user, ':', -1)
+            END),
+
+        NEW.user_relation = (CASE
+                                 WHEN position('#' in NEW._user) > 0 THEN split_part(NEW._user, '#', -1)
+                                 ELSE ''
+            END);
+END;
+$$ language 'plpgsql';
+-- +goose StatementEnd
+
+UPDATE tuple SET
+     user_object_type = (CASE
+                             WHEN position(':' in _user) > 0 THEN split_part(_user, ':', 1)
+                             ELSE ''
+         END),
+     user_object_id = (CASE
+                           WHEN position('#' in _user) > 0 THEN split_part(split_part(_user, ':', -1), '#', 1)
+                           ELSE split_part(_user, ':', -1)
+         END),
+     user_relation = (CASE
+                          WHEN position('#' in _user) > 0 THEN split_part(_user, '#', -1)
+                          ELSE ''
+         END);
+
+CREATE TRIGGER migrate_user_column
```

Comment:
It seems these triggers don't work like we expect.

Steps to reproduce the issue I am seeing:
1. Start Postgres
    `make start-postgres`
2. Migrate up to the latest schema version (version 3)
    `make migrate-postgres`
3. Run the latest version (v1.1.0) of OpenFGA locally
    `make run-postgres`
4. Write some tuples
5. Checkout this branch and run the migration for schema revision 4
    ```
    git checkout feat-split-user-column
    make migrate-postgres
    ```
6. Write another set of tuples


What I see is that the migration up to revision 4 (step 5) migrates existing tuples into the new user columns, but new writes (step 6) do not cause the `migrate_user_column_tuple` trigger. I believe the intent was for this trigger to cause updates to the new columns when values are written to the old columns (e.g. due to old OpenFGA replicas still being deployed and writing to the old columns).

Reply:
Hmm if I change this to `BEFORE INSERT` then step 6 fails with 

```
sql error: ERROR: assignment source returned 3 columns (SQLSTATE 42601)
```

Investigating..

Reply:
Ok fixed it :)